src/core/DataFile.cpp

```diff
@@ -584,34 +584,25 @@ bool DataFile::hasLocalPlugins(QDomElement parent /* = QDomElement()*/, bool fir
 
 DataFile::Type DataFile::type( const QString& typeName )
 {
-	for( int i = 0; i < TypeCount; ++i )
-	{
-		if( s_types[i].m_name == typeName )
-		{
-			return static_cast<DataFile::Type>( i );
-		}
-	}
+	const auto it = std::find_if(s_types.begin(), s_types.end(),
+		[&typeName](const TypeDescStruct& type) { return type.m_name == typeName; });
+	if (it != s_types.end()) { return it->m_type; }
 
 	// compat code
 	if( typeName == "channelsettings" )
 	{
-		return DataFile::InstrumentTrackSettings;
+		return Type::InstrumentTrackSettings;
 	}
 
-	return UnknownType;
+	return Type::Unknown;
 }
 
 
 
 
 QString DataFile::typeName( Type type )
 {
-	if( type >= UnknownType && type < TypeCount )
-	{
-		return s_types[type].m_name;
-	}
-
-	return s_types[UnknownType].m_name;
+	return s_types[static_cast<std::size_t>(type)].m_name;
```

Comment:
I believe so, yes. This function is only ever called with the value of `m_type`. This in turn is either set from the return value of `DataFile::type()`, which always returns a value in range, or from the value passed to the constructor, which is currently only called with enumerators of `Type`.

There are a couple of circumstances in which it could reasonably be out of range:
* Someone adds a new enumerator to `Type`, but doesn't add a corresponding entry to `s_types`. I would imagine this is unlikely, and would get caught quickly if it does happen - it's a fairly fundamental part of how this class works, and I would hope anyone adding types would be careful enough to do it properly.
* Someone constructs a `DataFile` with a type cast from some out-of-bounds value. This also seems unlikely, and would be better handled with an assertion in the `DataFile` constructor.